langgraph_workflow/utils/artifacts.py

```
@@ -0,0 +1,387 @@
+
+        Args:
+            thread_id: Thread identifier
+            artifact_type: Type of artifact (analysis, design, code, tests, etc.)
+            filename: Name of the file
+            content: Artifact content
+            metadata: Optional metadata to store
+
+        Returns:
+            Path to saved artifact
+        """
+        # Create type-specific subdirectory
+        type_dir = self.get_thread_dir(thread_id) / artifact_type
+        type_dir.mkdir(parents=True, exist_ok=True)
+
+        # Save main content
+        artifact_path = type_dir / filename
+        with open(artifact_path, "w", encoding="utf-8") as f:
+            f.write(content)
+
+        # Save metadata if provided
+        if metadata:
+            metadata_path = type_dir / f"{filename}.meta.json"
+            metadata_with_timestamp = {
+                **metadata,
+                "created_at": datetime.now().isoformat(),
+                "thread_id": thread_id,
+                "artifact_type": artifact_type,
+                "filename": filename,
+                "size_bytes": len(content.encode("utf-8")),
+            }
+
+            with open(metadata_path, "w", encoding="utf-8") as f:
+                json.dump(metadata_with_timestamp, f, indent=2)
+
+        logger.info(f"Saved artifact: {artifact_path}")
+        return artifact_path
+
+    def load_artifact(
+        self, thread_id: str, artifact_type: str, filename: str
+    ) -> Optional[str]:
+        """Load an artifact from the filesystem.
+
+        Args:
+            thread_id: Thread identifier
+            artifact_type: Type of artifact
+            filename: Name of the file
+
+        Returns:
+            Artifact content or None if not found
+        """
+        artifact_path = self.get_thread_dir(thread_id) / artifact_type / filename
+
+        if not artifact_path.exists():
+            logger.warning(f"Artifact not found: {artifact_path}")
+            return None
```

should we rather throw an exception here? If we expect an artifact to be there and it's not we should terminate.